logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageEnvelope.kt

```diff
@@ -0,0 +1,43 @@
+package com.wire.kalium.logic.data.message
+
+import com.wire.kalium.logic.data.conversation.ClientId
+import com.wire.kalium.logic.data.user.UserId
+
+data class MessageEnvelope(
```

Comment:
Note that this `MessageEnvelope` only makes sense for proteus.

Reply:
Yes. We can replace with an interface / sealed class later for different protocols.

Same should apply for a lot of the work being done now.